maps.Rmd

```diff
@@ -299,52 +299,140 @@ ggplot(dawson[-69]) +
-    min(mi_counties$lon), min(mi_counties$lat),
-    max(mi_counties$lon), max(mi_counties$lat)
-  )
-  mi_raster <- ggmap::get_openstreetmap(bbox, scale = 8735660)
-  saveRDS(mi_raster, "mi_raster.rds")
-}
+library(stars)
+sat_vis <- read_stars(img_vis, RasterIO = list(nBufXSize = 600, nBufYSize = 600))
+sat_inf <- read_stars(img_inf, RasterIO = list(nBufXSize = 600, nBufYSize = 600))
 ```
 
-Note that finding the appropriate `scale` in this example required a lot of manual tweaking, so it may take some effort to obtain the data you want. Once you have the raster data, the `ggmap()` function from the ggmap package allows you to create a plot drawing the map, as illustrated in the left panel below:
+In the code above, the first argument specifies the path to the raster file, and the `RasterIO` argument is used to pass a list of low-level parameters to GDAL. In this case, I have used `nBufXSize` and `nBufYSize` to ensure that R reads the data at low resolution (as a 600x600 pixel image). To see what information R has imported, we can inspect the `sat_vis` object:
 
 ```{r}
-ggmap::ggmap(mi_raster)
+sat_vis
+```
+
+This output tells us something about the structure of a stars object. For the `sat_vis` object the underlying data is stored as a three dimensional array, with `x` and `y` dimensions specifying the spatial data. The `band` dimension in this case corresponds to the colour channel (RGB) but is redundant for this image as the data are greyscale. In other data sets there might be bands corresponding to different sensors, and possibly a time dimension as well. Note that the spatial data is also associated with a coordinate reference system (referred to as "refsys" in the output). 
 
-mi_cities <- tbl_df(maps::us.cities) %>%
-  filter(country.etc == "MI") %>%
-  select(-country.etc, lon = long) 
+To plot the `sat_vis` data in ggplot2, we can use the `geom_stars()` function provided by the stars package. A minimal plot might look like this:
 
-ggmap::ggmap(mi_raster) +
-  geom_point(aes(size = pop), mi_cities, colour = "red") +
-  scale_size_area()
+`r columns(2, 1/2, max_width = 2)`
+```{r}
+ggplot() + 
+  geom_stars(data = sat_vis) + 
+  coord_equal()
 ```
 
-The nice thing about the `ggmap()` function is that the output is a ggplot object, so adding metadata to the plot is accomplished in the same way one would add layers to any other plot. This is illustrated in the right side plot above.
+The first thing to note is that `geom_stars()` requires the `data` argument to be a stars object, and its job is to take the stars object and render it using `geom_raster()` (the default) or `geom_sf()`. The choice of geom can be controlled by the user: you can set `sf = TRUE` to use `geom_sf()`, but note that rendering will be slow. he only way that `geom_stars()` is able to call `geom_sf()` this way is by converting the pixel map to a collection of polygons that `geom_sf()` understands, which is a much slower process than using `geom_raster()`. 
```

Comment:
Why would you ever want to use `geom_sf()`?

Reply:
My understanding is that if you need `geom_stars()` to rely on the CRS for anything, you have to go down the `geom_sf()` path rather than the `geom_raster()` path. Basing this off a pretty naive reading of the linked thread. Might be worth adding a note to explain why `geom_stars()` lets you do this?

https://github.com/r-spatial/stars/issues/21#issuecomment-461858810

Reply:
Proposed footnote here: f9e5235682205135b338f94383a50f938acd3525

Reply:
I'd think prefer to just avoid mentioning of `geom_sf()` here, in the possibly over optimistic expectation that the CRS problem will get resolved in the future